src/test/regress/expected/am_query.out

```diff
@@ -119,3 +121,49 @@ INSERT INTO union_second SELECT a, a FROM generate_series(11, 15) a;
 (10 rows)
 
 DROP TABLE union_first, union_second;
+-- https://github.com/citusdata/citus/issues/4600
+CREATE TABLE INT8_TBL_columnar(q1 int8, q2 int8) using columnar;
+INSERT INTO INT8_TBL_columnar VALUES('  123   ','  456');
+INSERT INTO INT8_TBL_columnar VALUES('123   ','4567890123456789');
+INSERT INTO INT8_TBL_columnar VALUES('4567890123456789','123');
+INSERT INTO INT8_TBL_columnar VALUES(+4567890123456789,'4567890123456789');
+INSERT INTO INT8_TBL_columnar VALUES('+4567890123456789','-4567890123456789');
+explain (costs off, summary off) select * from
+  INT8_TBL_columnar a left join lateral
```

Comment:
I guess it is also common to have lateral joins with `ON (true)` and have the join correlated in the subquery such as:

```SQL

SELECT COUNT(*) FROM t1 JOIN LATERAL (SELECT * FROM t2 WHERE t1.id = t2.id) as foo ON (true);
```

Postgres is probably smart enough to represent both the same, but still it might worth testing/adding test.